### PR TITLE
RDoc-1529 what is a document store

### PR DESCRIPTION
**Pull request by morhilai**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
-It acts as the connection manager and also exposes methods to perform all operations which you can run against an associated server instance.
+* The **Document Store** is the main entry point for the Client API.  
 
-The document store object has a list of URL addresses that points to RavenDB server nodes.
+* Establishes and manages the connection between a client application and a RavenDB server or cluster of servers.   
+  * Has a list of URL addresses that point to its associated server(s)  
+  * Accesses the server via HTTP requests  
 
-* `DocumentStore` acts against a remote server via HTTP requests, implementing a common `IDocumentStore` interface
+* Exposes the methods for performing all operations that can be run against the associated server(s).
```
Comment (Danielle9897) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: You have 'Exposes' here -and- in the last bullet - need to merge

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
+---
+{NOTE: }
 
-A document store is our main client API object which establishes and manages the connection channel between an application and a database instance. 
-It acts as the connection manager and also exposes methods to perform all operations which you can run against an associated server instance.
+* The **Document Store** is the main entry point for the Client API.  
 
-The document store object has a list of URL addresses that points to RavenDB server nodes.
+* Establishes and manages the connection between a client application and a RavenDB server or cluster of servers.   
+  * Has a list of URL addresses that point to its associated server(s)
```
Comment (Danielle9897) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: Redundant as 'cluster topology' is mentioned later

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
+# Client API: What is a Document Store
+---
+{NOTE: }
 
-A document store is our main client API object which establishes and manages the connection channel between an application and a database instance. 
-It acts as the connection manager and also exposes methods to perform all operations which you can run against an associated server instance.
+* The **Document Store** is the main entry point for the Client API.  
 
-The document store object has a list of URL addresses that points to RavenDB server nodes.
+* Establishes and manages the connection between a client application and a RavenDB server or cluster of servers.
```
Comment (Danielle9897) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: Above 2 bullets can/should be merged

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown [hunk 4]
```diff
@@ -1,21 +1,32 @@
...
+
+* The document store exposes the following Client API features:  
+  * [Session](../client-api/session/what-is-a-session-and-how-does-it-work)  
+  * [Operations](../client-api/operations/what-are-operations)  
+  * [Bulk insert](../client-api/bulk-insert/how-to-work-with-bulk-insert-operation)  
+  * [Changes API](../client-api/changes/what-is-changes-api)  
+  * [Conventions](../client-api/configuration/conventions)  
+  * [Events](../client-api/session/how-to/subscribe-to-events)  
+  * [Aggressive cache](../client-api/how-to/setup-aggressive-caching)  
+{NOTE/}
```
Comment (Danielle9897) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: **Replace** all above Note with:
=>

* The **Document Store** is the main Client API object that establishes and manages the communication between your client application and the [RavenDB cluster](.../add_link).  Communication is done via HTTP requests.

* The Document Store holds the Cluster Topology, the Authentication Certificate, and any configurations & customizations that you may have applied.

* Caching is built in. All requests to the server(s) and their responses are cached within the Document Store. 

* A single instance of a Document Store (singleton pattern) should be used per the lifetime of your application.

* The Document Store exposes methods to perform operations such as:

  * [Session](...) - Use the Session object to perform operations on a specific database
  * [Operations](...) - Mange the server with a set of low level operations commands 
  * [Bulk insert](...) - Useful when inserting a large amount of data
  * [Conventions](...) - Customize the Client API behaviour
  * [Changes API](...) - Receive messages from the server   
  * [Agressive Cache](...) - Configure caching behaviour
  * [Events](...) - Perform custom actions in response to Session's operations 
  * [Data Sucscriptions](...) - Define & manage data processing on the client side

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown (hunk 3 above)
Comment (morhilai) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: Fixed, see last comment

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown (hunk 2 above)
Comment (morhilai) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: Fixed, see last comment

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: Fixed

File: Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown [hunk 5]
```diff
@@ -1,21 +1,30 @@
...
+
+* The **Document Store** is the main Client API object which establishes and manages the communication between your client application and a [RavenDB cluster](../server/clustering/overview). 
+Communication is done via HTTP requests.  
+
+* The Document Store holds the [Cluster Topology](../server/clustering/rachis/cluster-topology), the [Authentication Certificate](../client-api/setting-up-authentication-and-authorization), 
+and any configurations & customizations that you may have applied.  
+
+* Caching is built in. All requests to the server(s) and their responses are cached within the Document Store.  
+
+* Only a single instance of the Document Store should be created per lifetime of your application ([Singleton Pattern](https://csharpindepth.com/articles/Singleton)).
```
Comment (Danielle9897) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: Note the addition of the 'per cluster'
No need for  'Only' 
=>
A single instance of a Document Store (singleton pattern) should be created per cluster per the lifetime of your application.

Comment (morhilai) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/what-is-a-document-store.dotnet.markdown]: done

**Pull request merged**